### PR TITLE
apply entity extension system for samples

### LINKED ISSUE #3361: Apply Entity Extension System for samples
See https://github.com/abpframework/abp/issues/3126#issuecomment-604053801

We can apply now for the samples depend on the framework as local reference.

For the samples referencing NuGet packages, we need to wait for v2.4 release. Created a separate issue for those samples: https://github.com/abpframework/abp/issues/3362

### PR DESCRIPTION
**Pull request by maliming**
And fix unit tests of EfCoreMigrationDemo project .

Resolve #3361

### CODE REVIEW

File: samples/BookStore-Angular-MongoDb/aspnet-core/src/Acme.BookStore.Domain/Users/AppUser.cs [hunk 1]
```diff
@@ -41,9 +41,19 @@ public class AppUser : FullAuditedAggregateRoot<Guid>, IUser
...
         /* Add your own properties here. Example:
-         *
-         * public virtual string MyProperty { get; set; }
-         */
+        *
+        * public string MyProperty { get; set; }
+        *
+        * If you add a property and using the EF Core, remember these;
+        *
+        * 1. update BookStoreDbContext.OnModelCreating
```
Comment (gdlcf88) [inline samples/BookStore-Angular-MongoDb/aspnet-core/src/Acme.BookStore.Domain/Users/AppUser.cs]: The `update` should be `Update`.

File: samples/BookStore-Modular/application/src/Acme.BookStore.Domain/Users/AppUser.cs [hunk 2]
```diff
@@ -41,9 +41,19 @@ public class AppUser : FullAuditedAggregateRoot<Guid>, IUser
...
         /* Add your own properties here. Example:
-         *
-         * public virtual string MyProperty { get; set; }
-         */
+        *
+        * public string MyProperty { get; set; }
+        *
+        * If you add a property and using the EF Core, remember these;
+        *
+        * 1. update BookStoreDbContext.OnModelCreating
```
Comment (gdlcf88) [inline samples/BookStore-Modular/application/src/Acme.BookStore.Domain/Users/AppUser.cs]: The `update` should be `Update`.

File: samples/BookStore/src/Acme.BookStore.Domain/Users/AppUser.cs [hunk 3]
```diff
@@ -41,9 +41,19 @@ public class AppUser : FullAuditedAggregateRoot<Guid>, IUser
...
         /* Add your own properties here. Example:
-         *
-         * public virtual string MyProperty { get; set; }
-         */
+        *
+        * public string MyProperty { get; set; }
+        *
+        * If you add a property and using the EF Core, remember these;
+        *
+        * 1. update BookStoreDbContext.OnModelCreating
```
Comment (gdlcf88) [inline samples/BookStore/src/Acme.BookStore.Domain/Users/AppUser.cs]: The `update` should be `Update`.

File: samples/DashboardDemo/src/DashboardDemo.Domain/Users/AppUser.cs [hunk 4]
```diff
@@ -41,9 +41,19 @@ public class AppUser : FullAuditedAggregateRoot<Guid>, IUser
...
         /* Add your own properties here. Example:
-         *
-         * public virtual string MyProperty { get; set; }
-         */
+        *
+        * public string MyProperty { get; set; }
+        *
+        * If you add a property and using the EF Core, remember these;
+        *
+        * 1. update BookStoreDbContext.OnModelCreating
```
Comment (gdlcf88) [inline samples/DashboardDemo/src/DashboardDemo.Domain/Users/AppUser.cs]: The `update` should be `Update`.

**Comment by maliming** [comment]
@hikalkan Done.

**Pull request merged**